### PR TITLE
feat: Setting of Routes Skeletons (REST API) (#39523)

### PR DESCRIPTION
**Pull request by Aamil-Successive**
Create a folder controllers inside src
Create folder user and trainee inside controllers.
Create following files in both folder
Controller.ts
routes.ts
index.ts

Define following routes in the trainee
get - /
post - /
put - /
delete - /

Define a class in Controller file having a method for each route returning some fake response.
Create a router.ts file in the root of src directory and import all the created routes and export them.
Import the router to the server file from router.ts file and use them in setUpRoutes method with /api prefix.

### CODE REVIEW

File: src/controllers/trainee/Controller.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  get = (req: Request, res: Response, next: NextFunction) => {
+    console.log('Get request..!');
+    res.status(200).send(trainees);
+  }
+
+  post = (req: Request, res: Response, next: NextFunction) => {
+    console.log('Post request..!');
+    console.log(req.body);
+    const trainee = {
+      traineeName: req.body.name,
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: use destructuring.

File: src/controllers/trainee/Controller.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    };
+    trainees.push(trainee);
+    res.status(200).send(trainee);
+  }
+  
+  put = (req: Request, res: Response, next: NextFunction) => {
+    console.log('Put request..!');
+    console.log(req.body);
+    console.log((req.params.id));
+    const trainee = trainees.find(e => e.id === parseInt(req.params.id, 10));
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: use destructuring for id

File: src/controllers/trainee/Controller.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if (!trainee) {
+      res.status(404).send('Not Found! Can not update your request!');
+    }
+    trainee.traineeName = req.body.name;
+    res.status(200).send(trainee);
+  }
+
+  delete = (req: Request, res: Response, next: NextFunction) => {
+    console.log('Delete request..!');
+    const trainee = trainees.find(e => e.id === parseInt(req.params.id, 10));
```
Comment (adilsuccessive) [inline src/controllers/trainee/Controller.ts]: use destructuring for id

File: src/controllers/user/Controller.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    get = (req: Request, res: Response, next: NextFunction) => {
+        console.log('Get request..!');
+        res.status(200).send(users);
+    }
+
+    post = (req: Request, res: Response, next: NextFunction) => {
+        console.log('Post request..!');
+        console.log(req.body);
+        const user = {
+            userName: req.body.name,
```
Comment (adilsuccessive) [inline src/controllers/user/Controller.ts]: use destructuring and then pass name

File: src/controllers/user/Controller.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        };
+        users.push(user);
+        res.status(200).send(user);
+    }
+
+    put = (req: Request, res: Response, next: NextFunction) => {
+        console.log('Put request..!');
+        console.log(req.body);
+        console.log((req.params.id));
+        const user = users.find(e => e.id === parseInt(req.params.id, 10));
```
Comment (adilsuccessive) [inline src/controllers/user/Controller.ts]: use destructuring

File: src/controllers/user/Controller.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        if (!user) {
+            res.status(404).send('Not Found! Can not update your request!');
+        }
+        user.userName = req.body.name;
+        res.status(200).send(user);
+    }
+
+    delete = (req: Request, res: Response, next: NextFunction) => {
+        console.log('Delete request..!');
+        const user = users.find(e => e.id === parseInt(req.params.id, 10));
```
Comment (adilsuccessive) [inline src/controllers/user/Controller.ts]: use destructuring

File: src/controllers/user/routes.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { Router } from 'express';
+import user from './Controller';
+
+const router: Router = Router();
+
+router.get('/', user.get);
+router.post('/', user.post);
+router.put('/:id', user.put);
```
Comment (adilsuccessive) [inline src/controllers/user/routes.ts]: pass id in request body instead of params for put route.

File: src/controllers/trainee/routes.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { Router } from 'express';
+import trainee from './Controller';
+
+const router: Router = Router();
+
+router.get('/', trainee.get);
+router.post('/', trainee.post);
+router.put('/:id', trainee.put);
```
Comment (adilsuccessive) [inline src/controllers/trainee/routes.ts]: pass id in request body instead of params for put route.

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

File: src/controllers/trainee/Controller.ts (hunk 1 above)
Comment (Aamil-Successive) [reply src/controllers/trainee/Controller.ts]: changes done

File: src/controllers/trainee/Controller.ts (hunk 2 above)
Comment (Aamil-Successive) [reply src/controllers/trainee/Controller.ts]: changes done

File: src/controllers/trainee/Controller.ts (hunk 3 above)
Comment (Aamil-Successive) [reply src/controllers/trainee/Controller.ts]: changes done

File: src/controllers/user/Controller.ts (hunk 4 above)
Comment (Aamil-Successive) [reply src/controllers/user/Controller.ts]: changes done

File: src/controllers/user/Controller.ts (hunk 5 above)
Comment (Aamil-Successive) [reply src/controllers/user/Controller.ts]: changes done

File: src/controllers/user/Controller.ts (hunk 6 above)
Comment (Aamil-Successive) [reply src/controllers/user/Controller.ts]: changes done

File: src/controllers/user/routes.ts (hunk 7 above)
Comment (Aamil-Successive) [reply src/controllers/user/routes.ts]: changes done

File: src/controllers/trainee/routes.ts (hunk 8 above)
Comment (Aamil-Successive) [reply src/controllers/trainee/routes.ts]: changes done

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**